### PR TITLE
go/proxyd: Add debug cache status header to HTTP responses

### PR DESCRIPTION
**Pull request by Inphi**
**Description**
Header is to be used only by proxyd devs for troubleshooting.

**Metadata**
- Fixes ENG-1896

### CODE REVIEW

**Comment by tynes** [comment]
Do you mind adding a changeset to this?

File: go/proxyd/server.go [hunk 1]
```diff
@@ -167,9 +168,10 @@ func (s *Server) HandleRPC(w http.ResponseWriter, r *http.Request) {
 				continue
 			}
 
-			batchRes[i] = s.handleSingleRPC(ctx, req)
```
Comment (tynes) [inline go/proxyd/server.go:170]: If I understand this correctly, whether or not the final request in the batch is cached will determine the caching status?

File: go/proxyd/server.go [hunk 2]
```diff
@@ -322,6 +325,15 @@ func (s *Server) populateContext(w http.ResponseWriter, r *http.Request) context
 	)
 }
 
+func setCacheHeader(w http.ResponseWriter, cached bool) {
+	const cacheStatusHdr = "X-Proxyd-Cache-Status"
```
Comment (tynes) [inline go/proxyd/server.go:329]: Would it be better to define this as a constant outside of this function?

File: go/proxyd/server.go (hunk 1 above)
Comment (Inphi) [reply go/proxyd/server.go:170]: meant that to have a single request in the batch to determine the caching status not the last. will fix.

File: go/proxyd/server.go (hunk 2 above)
Comment (Inphi) [reply go/proxyd/server.go:329]: hmm. yeah, makes it easy to discover.

**Pull request merged**